ui/apps/platform/src/Containers/Collections/CollectionsTable.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { Button, ButtonVariant, Truncate, Bullseye, Text } from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons';
+import { TableComposable, TableVariant, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+
+import EmptyStateTemplate from 'Components/PatternFly/EmptyStateTemplate';
+import LinkShim from 'Components/PatternFly/LinkShim';
+import useTableSelection from 'hooks/useTableSelection';
+import { CollectionResponse } from 'services/CollectionsService';
+import { collectionsPath } from 'routePaths';
+
+export type CollectionsTableProps = {
+    collections: CollectionResponse[];
+    hasWriteAccess: boolean;
+};
+
+function CollectionsTable({ collections, hasWriteAccess }: CollectionsTableProps) {
+    const history = useHistory();
+    const { selected, allRowsSelected, onSelect, onSelectAll } = useTableSelection(collections);
+
+    function onEditCollection(id: string) {
+        history.push({
+            pathname: `${collectionsPath}/${id}`,
+            search: 'action=edit',
+        });
+    }
+
+    function onCloneCollection(id: string) {
+        history.push({
+            pathname: `${collectionsPath}/${id}`,
+            search: 'action=clone',
+        });
+    }
+
+    return (
+        <>
+            <TableComposable variant={TableVariant.compact}>
+                <Thead>
+                    <Tr>
+                        {hasWriteAccess && (
+                            <Th
+                                select={{
+                                    onSelect: onSelectAll,
+                                    isSelected: allRowsSelected,
+                                }}
+                            />
+                        )}
+                        <Th modifier="wrap" width={25}>
+                            Collection
+                        </Th>
+                        <Th modifier="wrap">Description</Th>
+                        <Th modifier="wrap" width={10}>
+                            In use
+                        </Th>
+                        <Th aria-label="Row actions" />
+                    </Tr>
+                </Thead>
+                <Tbody>
+                    {collections.length === 0 && (
```

Comment:
Commented a few lines upstream to keep separate from thread below.

What are pro and con to replace “8 is enough” (could not resist the allusion) with something like:

```jsx
<Td colSpan={hasWriteAccess ? 3 : 5}>
```

Reply:
The only real con I can think of the possibility that a change down the line updates the number of columns and the empty state isn't well tested, leading to a mis-aligned UI.

According to the spec, this could be any value up to `1000` which we could use to simulate "full width". I did see some results online where this led to a massive page slow down in some versions of Firefox. Probably very unlikely, but I think I'd prefer an off-center error state to a page breaking edge case.